<launch>
    <!-- Default arguments -->
    <arg name="drone_id1" default="falcon"/>
    <arg name="drone_id2" default="$(arg drone_id1)1"/> 
    <arg name="drone_name_1" default="falcon"/>
    <arg name="drone_name_2" default="$(arg drone_name_1)1"/>
    <arg name="target_x" default="0.0"/>
    <arg name="target_y" default="0.0"/>
    <arg name="target_z" default="2.0"/>
    <arg name="target_x1" default="1.0"/>
    <arg name="target_y1" default="0.0"/>
    <arg name="target_z1" default="2.0"/>
    <arg name="kp" default="1.0"/>
    <arg name="ki" default="0.01"/>
    <arg name="kd" default="0.05"/>

    <!-- Node for Command-->
    <group ns="BEP">

        <node name="takeoff"
              pkg="drone_controller"
              type="takeoff_node"
              output="screen">
            <param name="drone_name_1" value="$(arg drone_name_1)" />
            <param name="drone_name_2" value="$(arg drone_name_2)" />
        </node>

        <node name="command_node_2_pos"
              pkg="drone_controller"
              type="command_node_2_pos"
              output="screen">
            <param name="drone_id1" value="$(arg drone_id1)"/>
            <param name="drone_id2" value="$(arg drone_id2)"/>
            <param name="target_x" value="$(arg target_x)"/>
            <param name="target_y" value="$(arg target_y)"/>
            <param name="target_z" value="$(arg target_z)"/>
            <param name="target_x1" value="$(arg target_x1)"/>
            <param name="target_y1" value="$(arg target_y1)"/>
            <param name="target_z1" value="$(arg target_z1)"/>
            <param name="kp" value="$(arg kp)"/>
            <param name="ki" value="$(arg ki)"/>
            <param name="kd" value="$(arg kd)"/>
        </node>
    </group>
</launch>
